pricegraph/src/orderbook/user.rs

```diff
@@ -30,8 +30,23 @@ impl User {
         order_id
     }
 
-    /// Retrieves the user's balance for a token
+    /// Return's the user's balance for the specified token.
     pub fn balance_of(&self, token: TokenId) -> f64 {
         self.balances.get(&token).copied().unwrap_or(0.0)
     }
+
+    /// Deducts an amount from the balance for the given token. Returns the new
+    /// balance or `None` if the user no longer has any balance.
+    pub fn deduct_from_balance(&mut self, token: TokenId, amount: f64) -> Option<f64> {
```

Comment:
Is this ok to do nothing if there is no or not enough balance?

Reply:
It should never happen since our capacities that we deduct always respect order limits and balances, but you are correct, there should be a debug assertion here.

Reply:
Added some `debug_assert!` statements to make sure its working as intended.